test/units/module_utils/common/text/test_converters.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2019, Andrew Klychkov @Andersson007 <aaklychkov@mail.ru>
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import json
+import pytest
+
+from datetime import datetime
+from pytz import timezone as tz
+
+from ansible.module_utils.common._collections_compat import Set
+from ansible.module_utils.common.text.converters import (
+    _json_encode_fallback,
+    container_to_bytes,
+    container_to_text,
+    jsonify,
+)
+
+
+DEFAULT_ENCODING = 'utf-8'
+
+
+class Test_json_encode_fallback:
+
+    """Namespace for testing _json_encode_fallback."""
```

Comment:
Usually, testing private interfaces doesn't make sense.

Reply:
it's not a method, just a common function

Reply:
but thank you for the note, will know

Reply:
It doesn't matter whether it's a method or a function. A private function is related to the module scope, a private method is related to the class. Still, both are private, it's just a different level of namespacing. If a module name starts with an underscore it'd be also private.

Reply:
ok, thank you for clarifying

Reply:
For algorithmic code, it can make sense to test private methods and private functions in isolation from the rest of the code.

This does seem to be a place where that could be justified.  The code being tested is functional (meaning it operates via parameters and return values rather than callbacks) and it plugs into a larger framework which is outside of our control.

What I'll sometimes do is push all the permutations of data that I care about at the private function and then push a small subset at the public interface to make sure that the interaction between the public and private code is working as expected.